File: grogroman/src/test/java/PlusMinusTest.java
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;
import java.util.List;
import java.util.stream.Collectors;

public class PlusMinusTest {

    @Test
    public void generalTest() {
        //given
        List<Integer> list = List.of(-4, 3, -9, 0, 4, 1);

        List<BigDecimal> expected = List.of(
                new BigDecimal("0.500000"),
                new BigDecimal("0.333333"),
                new BigDecimal("0.166667")
        );

        //when
        List<BigDecimal> actual = PlusMinus.plusMinus(list).collect(Collectors.toList());

        //then
        Assertions.assertEquals(expected, actual);
    }
}

File: grogroman/src/test/kotlin/io/violabs/grogroman/hackerrank/PlusMinusTest.kt
package io.violabs.grogroman.hackerrank

import io.violabs.grogroman.testEquals
import org.junit.jupiter.api.Test

class PlusMinusTest {

  @Test
  fun test() {
    // [-4, 3, -9, 0, 4, 1]
    //given
    val numbers = arrayOf(-4, 3, -9, 0, 4, 1)

    val expected: List<Double> = listOf(
      0.500000,
      0.333333,
      0.166667
    )

    //when
    val actual: Array<Double> = PlusMinus.run(numbers)

    //then
    expected.testEquals(actual)
  }
}